js/coinsbit.js

```diff
@@ -0,0 +1,483 @@
+                'cost': {
+                    'min': undefined,
+                    'max': undefined,
+                },
+            };
+            parsedMarketList.push ({
+                'id': id,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'active': isActive,
+                'precision': precision,
+                'limits': limits,
+                'info': market,
+            });
+        }
+        return parsedMarketList;
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const response = await this.publicGetTicker (this.extend ({ 'market': market['id'] }, params));
+        const ticker = this.safeValue (response, 'result');
+        return this.parseTicker (ticker, market);
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        await this.loadMarkets ();
+        const response = await this.publicGetTickers (params);
+        const tickers = this.safeValue (response, 'result');
+        const tickersIds = Object.keys (tickers);
+        const parsedTickers = [];
+        for (let tickerIndex = 0; tickerIndex < tickersIds.length; tickerIndex++) {
+            const tickerId = tickersIds[tickerIndex];
+            const tickerObject = tickers[tickerId];
+            const ticker = this.safeValue (tickerObject, 'ticker');
+            ticker['at'] = this.safeValue (tickerObject, 'at');
+            const market = this.markets_by_id[tickerId];
+            parsedTickers.push (this.parseTicker (ticker, market));
+        }
+        return this.filterByArray (parsedTickers, 'symbol', symbols);
+    }
+
+    parseTicker (ticker, marker) {
+        const timestamp = this.safeTimestamp (ticker, 'at', this.milliseconds ());
+        const datetime = this.iso8601 (timestamp);
+        const high = this.safeFloat (ticker, 'high');
+        const low = this.safeFloat (ticker, 'low');
+        const bid = this.safeFloat (ticker, 'bid');
+        const ask = this.safeFloat (ticker, 'ask');
+        const open = this.safeFloat (ticker, 'open');
+        const close = this.safeFloat (ticker, 'last');
+        const last = this.safeFloat (ticker, 'last');
+        const change = this.sum (last, -open);
+        let percentage = undefined;
+        if (open !== 0) {
+            percentage = parseFloat (change / open) * parseFloat (100);
+        }
+        const average = parseFloat (this.sum (last, open)) / parseFloat (2);
+        let baseVolume = undefined;
+        if ('vol' in ticker) {
+            baseVolume = this.safeFloat (ticker, 'vol');
+        } else if ('volume' in ticker) {
+            baseVolume = this.safeFloat (ticker, 'volume');
+        }
+        const quoteVolume = this.safeFloat (ticker, 'deal');
+        return {
+            'symbol': marker['symbol'],
```

Comment:
The `market` argument should have a default value of `undefined` in most of the parsers, so we need to derive the symbol from the ticker structure (or the trade structure, or the order structure, or whatever structure it is), rather than from the outside, as described here:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#parsing-symbols

Reply:
Thank you for comments. 
I don't have any information in ticker to identify a `symbol`, so i used market that i have in `fetchTicker`. `fetchTickers` has the same story but there a has a dictionary as a response with ids as keys, so i can find a market from those keys to provide it to the `parseTicker`. So can i use a `symbol` of `market` argument if `market` is not `undefined` like here?
```let symbol = undefined
if (market !== undefined) {
    symbol = market['symbol'];
}

Reply:
In that case we do something like the following in all the parsers to remain explicit (we are going to unify this under a base method for shorter syntax soon):

```JavaScript
function parseSomething (something, market = undefined) {
    let symbol = undefined;
    if (market !== undefined) {
        symbol = market['symbol'];
    }
    // ...
    return {
        // ...
        'symbol': symbol,
        // ...
    };
}

Reply:
For the above reason we also have to check the return values of `safe*` methods before actually trying arithmetic operations, otherwise that will raise an exception in most cases.